gapis/resolve/memory.go

```diff
@@ -85,7 +88,82 @@ func filterTypedRanges(ranges []*service.TypedMemoryRange) []*service.TypedMemor
 			newRanges = append(newRanges, ranges[i])
 		}
 	}
-	return newRanges
+	// Filter pnext pointer related structs, only keep the most specific one.
+	// When seeing a pointer struct, scan all the structs having the same root,
+	// and identify whether some are less precise and excessive structs caused by
+	// pnext pointer type castings. If found, filter them out.
+	toDelete := map[int]bool{}
+	index := 0
+	for index < len(newRanges) {
+		ty, err := types.GetType(newRanges[index].Type.TypeIndex)
+		if err != nil {
+			return nil, err
+		}
+		if strings.Contains(strings.ToLower(ty.Name), "void") {
+			root := newRanges[index].Root
+			dups := []int{}
+			for left := index - 1; left >= 0 && newRanges[left].Root == root; left-- {
+				dups = append(dups, left)
+			}
+			for right := index + 1; right < len(newRanges) && newRanges[right].Root == root; right++ {
+				dups = append(dups, right)
+			}
+			mostPreciseIndex := index
+			mostPreciseLevel, err := pnextPreciseLevel(newRanges[index].Type.TypeIndex)
+			if err != nil {
+				return nil, err
+			}
+			for _, i := range dups {
+				preciseLevel, err := pnextPreciseLevel(newRanges[i].Type.TypeIndex)
+				if err != nil {
+					return nil, err
+				}
+				if preciseLevel < 0 {
+					continue
+				} else if preciseLevel < mostPreciseLevel {
+					toDelete[i] = true
+				} else if preciseLevel > mostPreciseLevel {
+					toDelete[mostPreciseIndex] = true
+					mostPreciseIndex = i
+					mostPreciseLevel = preciseLevel
+				}
+			}
+			index = dups[len(dups)-1] + 1
+		} else {
+			index++
+		}
+	}
+	filteredRanges := []*service.TypedMemoryRange{}
+	for i := 0; i < len(newRanges); i++ {
+		if shouldDelete := toDelete[i]; !shouldDelete {
+			filteredRanges = append(filteredRanges, newRanges[i])
+		}
+	}
+	return filteredRanges, nil
+}
+
+// Return the pnext pointer derived struct's precise level.
```

Comment:
Go documentation requires the first sentence to start with the name of the thing being documented.

Reply:
Done in the new commit. Good to know that!